pcidevices update and introduction to sriov network devices (#390)
harvester/docs: docs/advanced/pcidevices.md

> ![](/img/v1.2/vm-import-controller/EnableAddon.png)

Reviewer: can we replace this img, thanks.

Author: done.

> Once addon is deployed successfully, it can take a few minutes for the controller to scan and PCIDevice crd's to be available.

Reviewer 2: ```suggestion
Once the `pcidevices-controller` addon is deployed successfully, it can take a few minutes for it to scan and the PCIDevice CRDs to become available.
```

Reviewer 2: Use `pcidevices-controller` to refer to the addon to stay consistent with the Harvester UI as well as the rest of this document, e.g., line 21

> Once the `pcidevices-controller` addon is deployed successfully, it can take a few minutes for it to scan and the PCIDevice CRDs to become available.

Reviewer 3: ```suggestion
Once the `pcidevices-controller` addon is deployed successfully, it can take a few minutes for it to scan and the PCI device CRDs to become available.
```

Should PCIDevice be replaced by PCI device?

Author: i should make it `PCIDevice` only to be consistent.

## Installing drivers for your PCI device inside the VM

This is just like installing drivers in the host. The PCI passthrough feature will bind the host device to the `vfio-pci` driver, which gives VMs the ability to use their own drivers. [Here is a screenshot](https://tobilehman.com/posts/suse-harvester-pci/#toc) of NVIDIA drivers being installed in a VM. It includes a CUDA example that proves that the device drivers work.

## SRIOV Network Devices
_Available as of v1.2.0_

![](/img/v1.2/pcidevices/SriovNetworkDevicesLink.png)

> The PCIDevices controller can now scan network interfaces on the underlying hosts and check if they support SRIOV Virtual Functions (VF's). If a valid device is found, the PCIDevices controller, will generate a new `SRIOVNetworkDevice` object.

Reviewer 2: ```suggestion
The `pcidevices-controller` addon can now scan network interfaces on the underlying hosts and check if they support SRIOV Virtual Functions (VFs). If a valid device is found, `pcidevices-controller` will generate a new `SRIOVNetworkDevice` object.
```

![](/img/v1.2/pcidevices/SriovNetworkDevicesList.png)

> A user can create Virtual Functions on a SriovNetworkDevice by choosing `Enable` and defining the `Number of Virtual Functions`

Reviewer 2: ```suggestion
To create VFs on a SriovNetworkDevice, you can click **⋮ > Enable** and then define the **Number of Virtual Functions**.
```

![](/img/v1.2/pcidevices/SriovNetworkDeviceEnable.png)

![](/img/v1.2/pcidevices/SriovNetworkVFDefinition.png)

> The PCIDevices controller will define the Virtual Functions on the Network Interface, and report the new PCIDevice status for the newly created Virtual Functions

Reviewer 2: ```suggestion
The `pcidevices-controller` will define the VFs on the network interface and report the new PCI device status for the newly created VFs.
```

![](/img/v1.2/pcidevices/SriovNetworkDevicesVFStatus.png)

> On the next re-scan, the `pcidevices-controller` will create the PCIDevices for VFs. This can take up to 1 minute.

Reviewer 3: ```suggestion
On the next re-scan, the `pcidevices-controller` will create the PCI devices for VFs. This can take up to 1 minute.
```
Should PCIDevice be replaced by PCI device?

Author: actually the CRD is `PCIDevice` only so that is correct.

> On the new re-scan by the PCIDevices controller, new PCIDevices for Virtual Functions. This can take upto 1 minute.

Reviewer 2: `On the new re-scan by the PCIDevices controller, new PCIDevices for Virtual Functions. ` is not a full sentence as it doesn't have a verb. Please refactor it so that it can be understood ^^
By the way, `upto` should be `up to`.

> Users can now navigate to the `PCI Devices` page, and view the new devices.

Reviewer 2: ```suggestion
You can now navigate to the **PCI Devices** page to view the new devices.
```

> A new filter has been introduced to assist users in filtering PCIDevices related to the underlying Network Interface.

Reviewer 2: ```suggestion
We have also introduced a new filter to help you filter PCI devices by the underlying network interface.
```

![](/img/v1.2/pcidevices/SriovNetworkDevicesFilter.png)

> The newly created PCIDevices can be passed through to Virtual Machines like any other PCIDevice.

Reviewer 2: ```suggestion
The newly created PCI device can be passed through to virtual machines like any other PCI device.
```
